app/authorities/parsers/academic_units_parser.rb

```diff
@@ -0,0 +1,15 @@
+module Parsers
+  class AcademicUnitsParser
+    def self.parse(jsonld)
+      extract_and_massage_data(jsonld)
+    end
+
+    private
+
+      def self.extract_and_massage_data(jsonld)
+        labels = JSON.parse(jsonld).dig('@graph').map { |graph| graph.dig("rdfs:label").dig("@value") unless graph.dig("rdfs:label").nil? }
```

Comment:
`#dig` takes an arbitrarily deep nested hash and returns nil if the any of the keys don't exist. Also, for readability sake I feel like this is better:

```
graph = JSON.parse(jsonld).dig("@graph")
raise "Did not find @graph in jsonld" unless graph
terms = graph.map { |g| { id: g.dig("@id"), label: g.dig("rdfs:label", "@value"), active: true } }
# return only terms that have id and label
terms.delete_if { |k,v| k.nil? || v.nil? } 
```